src/mongocxx/pool.cpp

```diff
@@ -39,20 +39,20 @@ void pool::_release(client* client) {
 
 pool::~pool() = default;
 
-pool::pool(const uri& uri, stdx::optional<options::ssl> ssl_options)
+pool::pool(const uri& uri, const options::client& options)
```

Comment:
Is it worth creating an options::pool type? Perhaps in the future the pool will be more configurable.